unit_testing/myapp/models.py

```diff
@@ -1,10 +1,44 @@
 from django.db import models
+from django.core.cache import cache
+from django.core.exceptions import ObjectDoesNotExist
 
 
 class Person(models.Model):
 
     first_name = models.TextField(max_length=50)
     last_name = models.TextField(max_length=50)
+    email = models.EmailField(unique=True)
+
+    cache_key = u'person_{}'
 
     def __unicode__(self):
-        return "{name}".format(name=self.first_name+self.last_name)
+        return "{fname} {lname}".format(fname=self.first_name, lname=self.last_name)
```

Comment:
use unicode values `u"{fname} {lname}"` and also use the strip function on this if the last name is empty.